File: March 2024 GFG SOLUTION/March-10.py
#User function Template for python3
class Solution:
    def removeDuplicates(self, str):
        # Initialize a set to keep track of seen characters
        seen = set()
        # Initialize an empty string to store the result
        result = ''
        
        # Iterate through each character in the string
        for char in str:
            # If the character has not been seen before, add it to the result and the set
            if char not in seen:
                result += char
                seen.add(char)
        
        return result

#{ 
# Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        str = input().strip()
        ob = Solution()
        ans = ob.removeDuplicates(str)
        print(ans)
        tc -= 1

# } Driver Code Ends

File: March 2024 GFG SOLUTION/March-10.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;
// } Driver Code Ends

class Solution {
public:
    std::string removeDuplicates(std::string str) {
        std::unordered_set<char> seen;
        std::string result = "";
        
        for (char& c : str) {
            if (seen.find(c) == seen.end()) {
                result += c;
                seen.insert(c);
            }
        }
        
        return result;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        string str;
        cin >> str;
        Solution ob;
        auto ans = ob.removeDuplicates(str);

        cout << ans << "\n";
    }
    return 0;
}
// } Driver Code Ends
